recipes/geotrans/all/conanfile.py

```diff
@@ -0,0 +1,95 @@
+from conans import ConanFile, CMake, tools
+import os
+import shutil
+
+
+class GeotransConan(ConanFile):
+    name = "geotrans"
+    version = "3.8"
+    license = (
+        "NGA GEOTRANS ToS (https://earth-info.nga.mil/php/download.php?file=wgs-terms)"
+    )
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://earth-info.nga.mil/"
+    description = "MSP GEOTRANS is the NGA and DOD approved coordinate converter and datum translator."
+    topics = (
+        "geotrans",
+        "geodesic",
+        "geographic",
+        "coordinate",
+        "datum",
+        "geodetic",
+        "conversion",
+        "transformation",
+    )
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+    generators = "cmake"
+    exports_sources = "CMakeLists.txt"
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(
+            **self.conan_data["sources"][self.version],
+            strip_root=True,
+            destination=self._source_subfolder,
+            filename="master.tgz"
+        )
+        shutil.copy("CMakeLists.txt", self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.configure(source_folder=self._source_subfolder)
+        return self._cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy(
+            "*.txt",
+            dst="licenses",
+            src=os.path.join(self._source_subfolder, "GEOTRANS3", "docs"),
+        )
+        self.copy("*", dst="res", src=os.path.join(self._source_subfolder, "data"))
+        self.copy(
+            "*.h", dst="include", src=os.path.join(self._source_subfolder, "CCS", "src")
+        )
+        self.copy("*.lib", dst="lib", src="lib", keep_path=False)
+        self.copy("*.dll", dst="bin", src="lib", keep_path=False)
+        self.copy("*.so", dst="lib", src="lib", keep_path=False)
+        self.copy("*.dylib", dst="lib", src="lib", keep_path=False)
+        self.copy("*.a", dst="lib", src="lib", keep_path=False)
+
+    def package_info(self):
+        self.cpp_info.resdirs = ["res"]
+        self.cpp_info.names["cmake_find_package"] = "geotrans"
+        self.cpp_info.names["cmake_find_package_multi"] = "geotrans"
+        self.cpp_info.components["dtcc"].names["cmake_find_package"] = "dtcc"
+        self.cpp_info.components["dtcc"].names["cmake_find_package_multi"] = "dtcc"
```

Comment:
We usually only set the cmake_find_package(_multi) names when the project's cmake script sets it.

Reply:
The project doesn't provide a cmake script unfortunately. This was my attempt at being as faithful to the way the library ships as possible - they provide two shared libraries, one of which can be used independently of the other. If this isn't the best way to provide this functionality, happy to make changes here.

Reply:
The current philosophy is to mimic the upstream project explicitly as best as possible.
When an upstream project does nothing, then we don't set anything explicitly and fall back to the conan defaults.

By using the conan defaults, cmake targets will still be generated for the components.

Reply:
Ahh, okay cool. Didn't realise they would still be generated.